### PR TITLE
Add option to set manage version and build number

### PR DESCRIPTION
**Pull request by shams-ahmed**
This PR adds a new input argument to resolve issues around managing Xcode 13 new version and build number input when exporting. 

By default, Xcode has this set to `true` which causes unintentional effects and hard to track down bugs with version numbers. This update allows step users to configure this option by setting the value. To help avoid more issues it's been set to `false`, similar to the behaviour of the Xcode archive step.

### CODE REVIEW

**Comment by ofalvai** [comment]
@shams-ahmed if it's not too much hassle, can you please to update go-steputils in this PR? 🥺 I want [this feature](https://github.com/bitrise-io/go-steputils/pull/42) to land in as many steps as possible.

File: bitrise.yml [hunk 1]
```diff
@@ -24,9 +24,16 @@ workflows:
         - branch: master
     - path::./:
         title: Step Test - iOS archive
+        is_always_run: true
```
Comment (ofalvai) [inline bitrise.yml:27]: Why is this needed?

File: bitrise.yml [hunk 2]
```diff
@@ -24,9 +24,16 @@ workflows:
...
         inputs:
         - export_method: development
         - archive_path: ./archives/ios.xcarchive
+    - path::./:
+        title: Step Test - iOS archive
```
Comment (ofalvai) [inline bitrise.yml:32]:
```suggestion
        title: Step Test - App Store archive
```

File: main.go [hunk 3]
```diff
@@ -318,11 +321,13 @@ func (s Step) Run(opts Config) (RunOut, error) {
 			return RunOut{}, fmt.Errorf("failed to write export options to file, error: %s", err)
 		}
 	} else {
-		exportOptionsContent, err := generateExportOptionsPlist(opts.ProductToDistribute, opts.DistributionMethod, opts.TeamID, opts.UploadBitcode, opts.CompileBitcode, opts.XcodebuildVersion.MajorVersion, archive)
+		exportOptionsContent, err := generateExportOptionsPlist(opts.ProductToDistribute, opts.DistributionMethod, opts.TeamID, opts.UploadBitcode, opts.CompileBitcode, opts.XcodebuildVersion.MajorVersion, archive, opts.ManageVersionAndBuildNumber)
 		if err != nil {
 			return RunOut{}, fmt.Errorf("failed to generate export options, error: %s", err)
 		}
 
+		log.Debugf("path %s", exportOptionsPath)
```
Comment (ofalvai) [inline main.go:329]: Is this useful though? The path is different in each build because it's inside the temp folder's random generated subfolder, so users won't be able to build a script based on the printed path.

File: step.yml [hunk 4]
```diff
@@ -198,6 +198,16 @@ inputs:
     - "no"
     is_required: true
 
+- manage_version_and_build_number: "no"
```
Comment (ofalvai) [inline step.yml:201]: Don't forget to run the `generate_readme` workflow to update the README with this new input!

File: step.yml [hunk 5]
```diff
@@ -198,6 +198,16 @@ inputs:
...
+  opts:
+    category: IPA export configuration
+    title: Manage version and build number
+    summary: This will change the version and build number of all content in your app. If your app has an invalid number (like one that was used previously, or precedes your current build number), the step provides an option to automatically increment it to a valid number. The step does not work if export options plist content has been set.
+    value_options:
+    - "yes"
+    - "no"
+    is_required: false
```
Comment (ofalvai) [inline step.yml:209]: I think this should be true if the input has a default value

File: step.yml [hunk 6]
```diff
@@ -198,6 +198,16 @@ inputs:
     - "no"
     is_required: true
 
+- manage_version_and_build_number: "no"
+  opts:
+    category: IPA export configuration
+    title: Manage version and build number
+    summary: This will change the version and build number of all content in your app. If your app has an invalid number (like one that was used previously, or precedes your current build number), the step provides an option to automatically increment it to a valid number. The step does not work if export options plist content has been set.
```
Comment (ofalvai) [inline step.yml:205]: It's not clear to me what the "yes" and "no" options mean based on this description. What does "manage" mean exactly, does Xcode increment the build number automatically?

File: main_test.go [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+	envRepository := env.NewRepository()
+	commandFactory := command.NewFactory(envRepository)
+	xcodebuildVersion, _ := utility.GetXcodeVersion(commandFactory)
+	archive, _ := xcarchive.NewIosArchive("configs.ArchivePath")
+
+	// When
+	result, _ := generateExportOptionsPlist("app", "development", "my team id", false, false, xcodebuildVersion.MajorVersion, archive, false)
+
+	// Then
+	if len(result) == 0 {
```
Comment (ofalvai) [inline main_test.go:24]: This could be easier expressed with functions from the `assert` package, for example:

```go
assert.Equal(t, 0, len(result))
```

You can also replace other assertions below

File: bitrise.yml (hunk 1 above)
Comment (shams-ahmed) [reply bitrise.yml:27]: This is to make sure the next step is run. currently, the test step fails because of provisioning profile issues on my machine. works fine on the ci

File: bitrise.yml (hunk 2 above)
Comment (shams-ahmed) [reply bitrise.yml:32]: updated

File: main.go (hunk 3 above)
Comment (shams-ahmed) [reply main.go:329]: removed log

File: step.yml (hunk 4 above)
Comment (shams-ahmed) [reply step.yml:201]: This PR #34 has made major changes to the readme.. should I just update that once this pr is merged?

File: step.yml (hunk 5 above)
Comment (shams-ahmed) [reply step.yml:209]: updated

File: step.yml (hunk 4 above)
Comment (ofalvai) [reply step.yml:201]: The generator gets data from step.yml, so if you add a new input in this PR, you should update the README too in the same PR as a rule of thumb

File: main_test.go (hunk 7 above)
Comment (shams-ahmed) [reply main_test.go:24]: nice one.. updated

**Review by ofalvai — APPROVED** [review]

File: step.yml (hunk 6 above)
Comment (shams-ahmed) [reply step.yml]: Updated the description. let me know what you think

**Review by ofalvai — APPROVED** [review]

**Review by ofalvai — APPROVED** [review]
